File: Java/src/CombinationSumTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.*;

public class CombinationSumTest {

    @Test
    public void testCombinationSum() {
        CombinationSum cs = new CombinationSum();

        List<List<Integer>> list0 = cs.combinationSum(new int[]{2, 3, 6, 7}, 7);
        Set<List<Integer>> set = new HashSet<>();
        set.add(Collections.singletonList(7));
        set.add(Arrays.asList(2, 2, 3));
        Assert.assertTrue(list0.size() == 2);
        Assert.assertEquals(new HashSet<>(list0), set);

        List<List<Integer>> list1 = cs.combinationSum(new int[]{1}, 3);
        set.clear();
        set.add(Arrays.asList(1, 1, 1));
        Assert.assertTrue(list1.size() == 1);
        Assert.assertEquals(new HashSet<>(list1), set);

        List<List<Integer>> list2 = cs.combinationSum(new int[]{1, 2}, 4);
        set.clear();
        set.add(Arrays.asList(1, 1, 1, 1));
        set.add(Arrays.asList(1, 1, 2));
        set.add(Arrays.asList(2, 2));
        Assert.assertTrue(list2.size() == 3);
        Assert.assertEquals(new HashSet<>(list2), set);
    }

}

File: Kotlin/src/CombinationSumTest.kt
import org.junit.Test

class CombinationSumTest {

    @Test
    fun testCombinationSum() {
        val cs = CombinationSum()

        val list0 = cs.combinationSum(intArrayOf(2, 3, 6, 7), 7)
        assert(list0.size == 2)
        assert(list0.containsAll(listOf(listOf(7), listOf(2, 2, 3))))

        val list1 = cs.combinationSum(intArrayOf(1), 3)
        assert(list1.size == 1)
        assert(list1[0] == listOf(1, 1, 1))

        val list2 = cs.combinationSum(intArrayOf(1, 2), 4)
        assert(list2.size == 3)
        assert(list2.containsAll(listOf(listOf(2, 2), listOf(1, 1, 2), listOf(1, 1, 1, 1))))
    }

}